packages/react-components/react-utilities/src/compose/getIntrinsicElementProps.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { getNativeElementProps } from '../utils/getNativeElementProps';
+import type { PropsWithObscureRef, UnknownSlotProps } from './types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type HTMLAttributes = React.HTMLAttributes<any>;
+
+/**
+ * Given an element tagname and user props, filters the props to only allowed props for the given
+ * element type.
+ *
+ * Equivalent to {@link getNativeElementProps}, but more type-safe.
+ */
+export const getIntrinsicElementProps = <
+  Props extends UnknownSlotProps,
+  ExcludedPropKeys extends Extract<keyof Props, string> = never,
+>(
+  /** The slot's element type (e.g. 'div') */
+  tagName: NonNullable<Props['as']>,
+  /** The component's props object */
+  props: PropsWithObscureRef<Props>,
```

Comment:
Because `ref` is another leak of bad typing and this is a good way to mitigate it. Since we remove `ref` from props declaration by using `ComponentProps` anyone can declared any `ref` and we're never verifying it:

<img width="895" alt="image" src="https://github.com/microsoft/fluentui/assets/5483269/3a9cf37c-ee9b-44d3-a1b0-580590056e3e">

So yeah, it's not necessary for internal functioning of the method, but it does help on getting wrong `ref` declarations:

<img width="895" alt="image" src="https://github.com/microsoft/fluentui/assets/5483269/fce9a1af-bc6f-4d28-81b5-3eb02538face">